.github/workflows/typescript-ci.yaml

```diff
@@ -0,0 +1,187 @@
+name: Protobuf CI
+
+on:
+  push:
+    paths:
+      - '**/*.proto'
+    branches:
+      - '**'
+  pull_request:
+    paths:
+      - '**/*.proto'
+  workflow_dispatch:
+
+jobs:
+  generate_protobuf:
+    name: Generate Protobuf Definitions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      
+      - name: Cache Protobuf Dependencies
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-protobuf-cache-${{ hashFiles('**/package-lock.json') }}
+      
+      - name: Install Dependencies
+        run: |
+          npm ci
+      
+      - name: Generate Protobuf Definitions
```

Comment:
We already have a script that generates protobufs, why are we forking a second process to do the same?